### PR TITLE
Added first test, tests support.

### PR DESCRIPTION
**Pull request by pp-mo**
Add some basic testing support and one test, supposedly working...

### CODE REVIEW

File: iris_grib/__init__.py [hunk 1]
```diff
@@ -36,11 +37,16 @@
...
 import scipy.interpolate
+import threading
+import warnings
 
 import iris
 import iris.fileformats.rules as iris_rules
-from iris._deprecation import warn_deprecated
-from iris.analysis._interpolate_private import Linear1dExtrapolator
+try:
+    from iris.analysis._interpolate_private import Linear1dExtrapolator
```
Comment (pelson) [inline iris_grib/__init__.py]: We *definitely* need a comment here if it is to remain in the PR...

**Comment by pelson** [comment]
Tempted to merge even though `iris_grib.FUTURE.strict_grib_load` is not necessarily going to be the necessary mechanism.

**Comment by rhattersley** [comment]
> Tempted to merge even though iris_grib.FUTURE.strict_grib_load is not necessarily going to be the necessary mechanism.

Let's ditch the whole FUTURE thing.

**Comment by pp-mo** [comment]
Now works again, drastically reduced, hopefully now complies with "Let's ditch the whole FUTURE thing."

This currently defines it's own placeholder for 'warn_deprecated', as it doesn't exist in Iris 1.9.
That can be **removed** when we have ditched all the deprecated functionality.
Ok @rhattersley ?

**Comment by rhattersley** [comment]
Close enough. Let's make progress...

**Pull request merged**